src/overview/overview.md

```diff
@@ -0,0 +1,643 @@
+        label="Runtime Dialect"
+        graph[style=bold, bgcolor="darkseagreen1", labeljust=l]
+        runtime_initial [label="Initial"]
+        runtime_post_cleanup [label="Post Cleanup"]
+        runtime_optimized [label="Optimized"]
+        runtime_initial -> runtime_post_cleanup
+        runtime_post_cleanup -> runtime_optimized
+    }
+
+    subgraph cluster_legend {
+        rankdir=LR
+        node [shape=plaintext, width=0, height=0.5]
+        label = "Legend";
+        {rank=same; key, key2 }
+        key [label=<<table border="0" cellpadding="2" cellspacing="0" cellborder="0">
+        <tr><td align="right" port="i1">Transformation </td></tr>
+        </table>>]
+        key2 [label=<<table border="0" cellpadding="2" cellspacing="0" cellborder="0">
+        <tr><td port="i1"> </td></tr>
+        </table>>]
+        key:i1:e -> key2:i1:w [color="blue"]
+    }
+
+    labelloc=b
+    label="Mid-level Intermediate Representation (MIR) Dialects and Phases"
+}
+```
+
+MIR optimizations generally are opportunistic: because we're before
+monomorphization, there are e.g. constants related to generic parameters whose
+value we cannot know. When an opportunity does arise, however, MIR optimizations
+try to achieve one or both of the following objectives:
+
+1. to reduce the amount of work for the codegen backends, and
+2. to produce simpler or more optimizable backend output.
+
+See [Mid-level Intermediate
+Representation](https://rustc-dev-guide.rust-lang.org/mir/index.html) for more
+details.
+
+#### Typed High-level Intermediate Representation (THIR) Building and Analysis
+
+Typed High-level Intermediate Representation (THIR) is a further desugared
+version of High-level Intermediate Representation (HIR) with full type
+information embedded. THIR is the last IR where lexical structure is meaningful
+for analysis in the chain of lowering from HIR to THIR to MIR and beyond, where
+lower-level IRs transition from being lexically-aware to being dataflow-aware.
+For example, unsafeck is performed on THIR and not MIR because we care about the
+lexical structure -- whether something exists inside an `unsafe` block
+lexically. To further simplify tree-based reasoning, THIR has several
+simplifications over HIR:
+
+- Explicit auto-deref/auto-ref and coercions.
+- Method calls are desugared into regular function calls.
+- Overloaded operators like `+` are lowered into function calls or builtin
+  operations.
+- Match ergonomics are desugared.
+
+With the aforementioned properties, the construction of THIR synthesizes enough
+information for MIR building, and makes lowering to MIR easier and less
+error-prone than if HIR was attempted to be directly lowered to MIR.
+
+See [The Typed High-level Intermediate
+Representation](https://rustc-dev-guide.rust-lang.org/thir.html) for more
+details.
+
+#### HIR Body Type-checking
+
+> 🚧 **TODO** 🚧 (Remove the cat meme when complete with draft)
+
+![](https://hackmd.io/_uploads/ry2CM6gAa.jpg)
```

Comment:
:rofl: plz don't remove the cat meme

Reply:
something something retaining the cat meme might require an FCP in the final version /jk